src/AspNetCore.Mvc.HttpActionResults.ClientError/RequestedRangeNotSatisfiableResult.cs

```diff
@@ -0,0 +1,52 @@
+﻿namespace Microsoft.AspNetCore.Mvc
+{
+    using Http;
+    using Net.Http.Headers;
+    using System;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// A <see cref="StatusCodeResult"/> that when executed will produce an empty
+    /// <see cref="StatusCodes.Status416RequestedRangeNotSatisfiable"/> response.
+    /// </summary>
+    public class RequestedRangeNotSatisfiableResult : StatusCodeResult
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RequestedRangeNotSatisfiableResult"/> class.
+        /// </summary>
+        public RequestedRangeNotSatisfiableResult()
+            : base(StatusCodes.Status416RequestedRangeNotSatisfiable)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RequestedRangeNotSatisfiableResult"/> class.
+        /// </summary>
+        /// <param name="selectedResourceLength"></param>
+        public RequestedRangeNotSatisfiableResult(long selectedResourceLength)
+            : this()
+        {
+        }
+
+        /// <summary>
+        /// Gets or sets the current length of the selected resource.
+        /// </summary>
+        public long SelectedResourceLength { get; set; }
```

Comment:
Can you use nullable long instead of one with 0 default value? It is the way the MVC team do it:
https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/ObjectResult.cs#L32